src/main/java/maksym/fedorenko/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,67 @@
+package maksym.fedorenko.bookstore.exception;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex, HttpHeaders headers,
+            HttpStatusCode status, WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```

Comment:
let's enhance this handler :)

1. let's gather all the errors, along with their fields into a map
2. let's retrieve URI of the endpoint, these error related to
3. let's log that

```
Map<String, String> errors = new HashMap<>();
ex.getBindingResult().getAllErrors()
     .forEach(e -> errors.put(((FieldError) e).getField(), e.getDefaultMessage()));

String requestUri = ((ServletWebRequest) request).getRequest().getRequestUri();

logger.debug("Validation errors for {}: {}", requestUri, errors); // you need to add logger - this can easily be done with Lombok

return new ErrorResponseWrapper("bad-request", "Request input parameters are missing or invalid"); // see my comment regarding this wrapper 
```